var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1242,6 +1267,35 @@ def setup_dependent_package(self, module, dependent_spec):
         module.python_platlib = join_path(dependent_spec.prefix, self.platlib)
         module.python_purelib = join_path(dependent_spec.prefix, self.purelib)
 
+    def add_files_to_view(self, view, merge_map, skip_if_exists=True):
+        """Make the view a virtual environment if it isn't one already.
+
+        If `python-venv` is linked into the view, it will already be a virtual
+        environment. If not, then this is an older python that doesn't use the
+        python-venv support, or we may be using python packages that
+        use ``depends_on("python")`` but not ``extends("python")``.
+
+        We used to copy the python interpreter in, but we can get the same effect in a
+        simpler way by adding a ``pyvenv.cfg`` to the environment.
+
+        """
+        super().add_files_to_view(view, merge_map, skip_if_exists=skip_if_exists)
+
+        # location of python inside the view, where we will put the venv config
+        projection = view.get_projection_for_spec(self.spec)
+        pyvenv_cfg = os.path.join(projection, "pyvenv.cfg")
+        if os.path.lexists(pyvenv_cfg):
+            return
+
+        # if this is a copy view, the realpath of python will be in the view
+        python_path = find_python_in_prefix(self.spec, projection).path
```

Comment:
I think it would be slightly better to "find" it in `merge_map`, since that means this package has to add it to the view.

Reply:
Right now we link sequentially, in a pretty sensible order. But yes it would be nice to be able to parallelize that. I don't see a way without rearchitecting how merge maps are done, though.

For this one you really want to know whether some other package will link a particular file into a view. I think the package that wants to know (here, python) would need to say so in advance and then when you construct the merge map you can fill in additional details to be passed in to `add_files_to_view`.  I don't see any other way to do it without computing everything in advance and passing all that to every merge... which seems like too much (and also likely limits parallelism).